ipn/ipnlocal/local.go

```diff
@@ -1635,6 +1635,38 @@ func (b *LocalBackend) authReconfig() {
 				set(peer.Name, peer.Addresses)
 			}
 		}
+
+		addFallback := func() {
+			for _, resolver := range nm.DNS.FallbackResolvers {
+				res, err := parseResolver(resolver)
+				if err != nil {
+					b.logf(err.Error())
+					continue
+				}
+				dcfg.DefaultResolvers = append(dcfg.DefaultResolvers, res)
+			}
+		}
+
+		// Set FallbackResolvers as the default resolvers in the
+		// scenarios that can't handle a purely split-DNS config. See
+		// https://github.com/tailscale/tailscale/issues/1743 for
+		// details.
+		switch {
+		case len(dcfg.DefaultResolvers) != 0:
+			// Default resolvers already set.
+		case len(dcfg.Routes) == 0 && len(dcfg.Hosts) == 0 && len(dcfg.AuthoritativeSuffixes) == 0:
+			// No settings requiring split DNS, no problem.
+		case (version.OS() == "iOS" || version.OS() == "macOS") && !uc.ExitNodeID.IsZero():
```

Comment:
is this right for tailscaled-on-macOS? I don't think so?

Reply:
tailscaled-on-macos has no DNS management atm, right?

Do we have a helper to figure out standalone vs. netext/sysext?